### ISSUE TITLE
`mix deps.clean --all` fails on permission denied error

### ISSUE THREAD

**Issue by simonmcconnell** [elixir-lang/elixir#13060]
### Elixir and Erlang/OTP versions

```
Erlang/OTP 25 [erts-13.2.2.4] [source] [64-bit] [smp:48:32] [ds:48:32:10] [async-threads:1] [jit:ns]

Elixir 1.15.7 (compiled with Erlang/OTP 25)
```

### Operating system

Windows 11

### Current behavior

A dep intermittently fails to be cleaned.  It is not always the same dep.  Sometimes the command needs to be run several times to get through them all.

```
> mix deps.clean --all
...
* Cleaning telemetry
* Cleaning telemetry_metrics
** (File.Error) could not remove files and directories recursively from "r:/code/logacy/logacy/deps/telemetry_metrics": permission denied
    (elixir 1.15.7) lib/file.ex:1356: File.rm_rf!/1
    (elixir 1.15.7) lib/enum.ex:984: Enum."-each/2-lists^foreach/1-0-"/2
    (mix 1.15.7) lib/mix/tasks/deps.clean.ex:66: Mix.Tasks.Deps.Clean.run/1
    (mix 1.15.7) lib/mix/task.ex:455: anonymous fn/3 in Mix.Task.run_task/5
    (mix 1.15.7) lib/mix/task.ex:514: Mix.Task.run_alias/6
    (mix 1.15.7) lib/mix/cli.ex:92: Mix.CLI.run_task/2
    c:/Users/simon/scoop/apps/elixir/current/bin/mix:2: (file)
```

run it again

```
mix deps.clean --all                                                                                                                                  ─╯
* Cleaning telemetry_metrics
warning: the dependency telemetry_metrics is not present in the build directory
* Cleaning telemetry_poller
...
```

### Expected behavior

cleans without errors :)

**Comment by josevalim** [comment]
Do you have a version of the app running? Or have all instances been killed before running?

**Comment by simonmcconnell** [comment]
The only erl.exe process I can see is the one under vscode.  

I tested it a few times last night without the app running and I couldn't get it to fault.  I tried opening vscode and doing it but I couldn't get it to fault.  However, I was running the app in vscode terminal this morning.  When I was ready to release, I killed it, tried to run my release mix task (which calls mix deps.clean --all) and it failed.

I doubt there is anything we can do about this other than don't run it when the editor is open...?

**Comment by josevalim** [comment]
Correct, not much to do, but we can replace a hard raise by a logged error, so at least we delete the most we can on every run. WDYT?

**Comment by simonmcconnell** [comment]
Sounds good.  We could collect the ones that fail and try them a second time, then report whatever wasn't removed after trying twice.

**Comment by josevalim** [comment]
PRs are welcome!

**Comment by michallepicki** [comment]
This can probably be closed now, after #13161 got merged
